README.asciidoc

```diff
@@ -0,0 +1,408 @@
+* asyncio - for referee commands client 
+* serial - needed for coding usb communication
+* struct - for packing information that is sent to the motherboard
+* serial.tools.list_ports - to check for available usb ports, so we don't have to find the correct port's name manually
+* jit_functions - used it in image processing, jit makes for loops much faster so we can go through lists much faster
+* numpy - for list manipulation
+* cv2 - image processing
+* tkinter - used it in the beginning to simulate omnimotion visually
+* websockets - to make our testing server for referee commands
+
+=== A brief description of your game logic and overall code structure.
+
+Our code overall is section into files according to their usage (for example image_processing.py and movement.py) to make the code less cluttered then if it was all in the same file. The fails contain mainly different classes and then we can import them according to our needs. The main code is structured so that every situation has a State that we can refer to to make the robot move accordingly.
+
+=== Block diagram of game logic
+image::Images/diagramm2.JPG[1000,1000]
+
+=== Good and bad
+
+* One of our code's strong side is sigmoid controller. It eliminated most of the if else sentences and also made the robot's movement better.
+* Ball detection and depth dedection could be improved by using better and more calculations.
+* Timer that we used could be made into a function to make the code more compact.
+* Line dedection is very reliable.
+* Code was good enough to win DeltaX.
+
+
+== Electronics
+
+The electronics of the robot use the components that were recommended and provided by the instructors.
+The mainboard can control 3 motors and 2 servos. The motor drivers are powered by a 4S LiPo battery and the microcontroller side is powered through the usb connector by the NUC computer. The USB connection is also used to communicate between the microcontroller and the computer. There is a connector for an IR sensor to detect the ball in the holder. The microcontroller will let the computer know when the IR sensor detects a ball.
+
+=== Some problems that occurred and the final outcome
+* After soldering all the components, there was a problem with the 5V on the battery side of the board. It turned out that one of the pins of the switching regulator was not connected.
+* Some of the PWM and DIR pins connections were swapped. Needed to hack the board to fix it.
+* Micro USB connector broke and had to be replaced.
+* IR sensor wire broke during the robot assembly and it needed to be fixed.
+
+* Everything worked in the end and we won Delta-X and the course competition.
+
+image::Images/mainboard1.jpg[500,500]
+image::Images/mainboard2.jpg[500,500]
+image::Images/robot.jpg[500,500]
+image::Images/mainboard_with_wires.jpg[950,500]
+
+== Mechanics
+
+=== Thrower design
+* Ball grabbing ability 
+
+image:Images/thrower1.png[] 
+
+* Adjustable throwing angle 

```

Comment:
A short description of what was used to adjust it would also help. If one knows the design, it is easy to see how it works but without prior knowledge, a picture and one line of deception can be hard to understand.

Overall grabbing and potential angle adjustment are non-standard and should be mentioned somewhat.

Reply:
Added description under the image:  Adjustable throwing angle is achieved by rotating a gear wheel attached to a servo, which rises or lowers the ramp